src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -110,6 +113,22 @@ public static class DependencyGraphRestoreUtility
             return new List<RestoreSummary>();
         }
 
+        private static void PersistDGSpec(DependencyGraphSpec dgSpec)
+        {
+            var filePath = Path.Combine(
+                    NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp),
+                    "nuget-dg",
+                    "nugetSpec.dg");
+
+            // delete nugetSpec.dg if exists
+            FileUtility.Delete(filePath);
+
+            // create nuget temp folder if not exists
+            DirectoryUtility.CreateSharedDirectory(Path.GetDirectoryName(filePath));
+
+            dgSpec.Save(filePath);
```

Comment:
This needs to use `FileUtility.Replace` instead of deleting and then saving directly to the output path. Replace will write to a temp file first.

Reply:
@emgarten feedback client tool can't access solution or project folder or anything from project which is why it will require it to be the unique file which they can always try to access and capture if available. 
I agree there is potential to get wrong dg spec but i dont think it will be a common case since most scenarios, customers repro's a issue and file feedback. But again, we can start with this (since it's easy to get integrate) and see, if there are more negative cases then we can go another route which will be to provide an VS ide service to capture all these details.

Reply:
yeah it makes more sense!